File: src/main/java/net/dustley/clean_cut/block/ModBlocks.java
package net.dustley.clean_cut.block;

import net.dustley.clean_cut.CleanCut;
import net.dustley.clean_cut.block.custom.FleshBlock;
import net.dustley.clean_cut.block.custom.ForgorFleshBlock;
import net.dustley.clean_cut.block.custom.TendonChain;
import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents;
import net.minecraft.block.*;
import net.minecraft.block.piston.PistonBehavior;
import net.minecraft.item.BlockItem;
import net.minecraft.item.Item;
import net.minecraft.registry.Registries;
import net.minecraft.registry.Registry;
import net.minecraft.sound.BlockSoundGroup;
import net.minecraft.util.Identifier;

import static net.dustley.clean_cut.item.ModItems.CLEAN_CUT_ITEM_GROUP_KEY;

public class ModBlocks {

    public static final Block FLESH = registerBlock("flesh",
            new FleshBlock(AbstractBlock.Settings.create().sounds(BlockSoundGroup.HONEY)));
/*
    public static final Block FLESH_STAIRS = registerBlock("flesh_stairs",
            new StairsBlock(ModBlocks.FLESH_STAIRS.getDefaultState(),AbstractBlock.Settings.create().sounds(BlockSoundGroup.HONEY)));
    public static final Block FLESH_SLAB = registerBlock("flesh_slab",
            new SlabBlock(AbstractBlock.Settings.create().sounds(BlockSoundGroup.HONEY)));
*/




    public static final Block FORGOTTEN_FLESH = registerBlock("forgotten_flesh",
            new ForgorFleshBlock(AbstractBlock.Settings.create().sounds(BlockSoundGroup.HONEY).jumpVelocityMultiplier(-0.05f)));



    public static final Block CARRION_BONE_BLOCK = registerBlock("carrion_bone_block",
            new PillarBlock(AbstractBlock.Settings.create().sounds(BlockSoundGroup.BONE).requiresTool().strength(2.0F)));
    public static final Block CARRION_BONE_BUNDLE = registerBlock("carrion_bone_bundle",
            new PillarBlock(AbstractBlock.Settings.create().sounds(BlockSoundGroup.BONE).requiresTool().strength(2.0F)));



    public static final Block DENTED_LIVING_STEEL = registerBlock("dented_living_steel",
            new Block(AbstractBlock.Settings.create().sounds(BlockSoundGroup.COPPER).requiresTool().strength(3.0F, 6.0F)));



    public static final Block COMPACTED_LIVING_STEEL = registerBlock("compacted_living_steel",
            new Block(AbstractBlock.Settings.create().sounds(BlockSoundGroup.COPPER).requiresTool().strength(3.0F, 6.0F)));
    public static final Block FRAMED_LIVING_STEEL = registerBlock("framed_living_steel",
            new Block(AbstractBlock.Settings.create().sounds(BlockSoundGroup.COPPER).requiresTool().strength(3.0F, 6.0F)));
    public static final Block SEETHING_LANTERN = registerBlock("seething_lantern",
            new LanternBlock(AbstractBlock.Settings.create().sounds(BlockSoundGroup.LANTERN).requiresTool().strength(3.5F).sounds(BlockSoundGroup.LANTERN).luminance((state) -> {
                return 15;
            }).nonOpaque().pistonBehavior(PistonBehavior.DESTROY).solid()));
    public static final Block TENDON_CHAIN = registerBlock("tendon_chain",new TendonChain(AbstractBlock.Settings.create().solid().requiresTool().strength(5.0F, 6.0F).sounds(BlockSoundGroup.CHAIN).nonOpaque()));





    private static Block registerBlock(String name, Block block) {
        registerBlockItem(name, block);
        ItemGroupEvents.modifyEntriesEvent(CLEAN_CUT_ITEM_GROUP_KEY)
                .register(entries -> {
                entries.add(block);

                });
        return Registry.register(Registries.BLOCK, Identifier.of(CleanCut.MOD_ID, name), block);
    }

    private static void registerBlockItem(String name, Block block) {
        Registry.register(Registries.ITEM, Identifier.of(CleanCut.MOD_ID, name),
                new BlockItem(block, new Item.Settings()));
    }

    public static void registerModBlocks() {
        CleanCut.LOGGER.info("Registering Mod Blocks for " + CleanCut.MOD_ID);


    }
}

File: src/main/kotlin/net/dustley/clean_cut/block/ModBlocks.kt
package net.dustley.clean_cut.block

import net.dustley.clean_cut.CleanCut
import net.dustley.clean_cut.block.lantern.SeethingLanternBlock
import net.fabricmc.fabric.api.itemgroup.v1.FabricItemGroupEntries
import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents
import net.minecraft.block.AbstractBlock
import net.minecraft.block.Block
import net.minecraft.block.MapColor
import net.minecraft.block.PillarBlock
import net.minecraft.block.enums.NoteBlockInstrument
import net.minecraft.block.piston.PistonBehavior
import net.minecraft.item.BlockItem
import net.minecraft.item.Item
import net.minecraft.item.ItemGroups
import net.minecraft.registry.Registries
import net.minecraft.registry.Registry
import net.minecraft.sound.BlockSoundGroup

object ModBlocks {

    val FLESH: Block = registerBlock("flesh", Block(AbstractBlock.Settings.create()
        .mapColor(MapColor.RED).instrument(NoteBlockInstrument.BASEDRUM).strength(3.0f, 4.0f).sounds(BlockSoundGroup.SLIME)))

    val CARRION_BONE: Block = registerBlock("carrion_bone", PillarBlock(AbstractBlock.Settings.create()
        .mapColor(MapColor.DULL_RED).instrument(NoteBlockInstrument.PIGLIN).requiresTool().strength(2.0f).sounds(BlockSoundGroup.BONE)))

    val COMPACTED_LIVING_STEEL: Block = registerBlock("compacted_living_steel", Block(AbstractBlock.Settings.create()
            .mapColor(MapColor.DULL_RED).instrument(NoteBlockInstrument.BASEDRUM).requiresTool().strength(5.0f, 6.0f).sounds(BlockSoundGroup.COPPER)))

    val DENTED_LIVING_STEEL: Block = registerBlock("dented_living_steel", Block(AbstractBlock.Settings.create()
        .mapColor(MapColor.BROWN).instrument(NoteBlockInstrument.BASEDRUM).requiresTool().strength(5.0f, 6.0f).sounds(BlockSoundGroup.COPPER)))

    val SEETHING_LANTERN: Block = registerBlock("seething_lantern",
        SeethingLanternBlock(
            AbstractBlock.Settings.create().mapColor(MapColor.IRON_GRAY).solid().requiresTool().strength(3.5f)
                .sounds(BlockSoundGroup.LANTERN).luminance { 13 }
                .nonOpaque().pistonBehavior(PistonBehavior.DESTROY)
        ))

    private fun registerBlockWithoutBlockItem(name: String, block: Block): Block {
        return Registry.register(Registries.BLOCK, CleanCut.identifier(name), block)
    }

    private fun registerBlock(name: String, block: Block): Block {
        registerBlockItem(name, block)
        return Registry.register(Registries.BLOCK, CleanCut.identifier(name), block)
    }

    private fun registerBlockItem(name: String, block: Block) {
        Registry.register(
            Registries.ITEM, CleanCut.identifier(name),
            BlockItem(block, Item.Settings())
        )
    }

    fun registerModBlocks() {
        CleanCut.LOGGER.info("Registering Mod Blocks for " + CleanCut.MOD_ID)

        ItemGroupEvents.modifyEntriesEvent(ItemGroups.BUILDING_BLOCKS)
            .register(ItemGroupEvents.ModifyEntries { entries: FabricItemGroupEntries ->
                entries.add(FLESH)
                entries.add(CARRION_BONE)
                entries.add(COMPACTED_LIVING_STEEL)
                entries.add(DENTED_LIVING_STEEL)
                entries.add(SEETHING_LANTERN)
            })
    }
}